Add search command for SSM backend

pecigonzalo opened the pull request:
This is a draft implementation of a search command.

In many cases we need to search for services and we have to fallback to `aws ssm` which is ok,
but it could be a good idea to implement this simple search functionality in `chamber`.


Let me know your thoughts and ill add the missing parts

### Example output
Service name: `servicename`
```
❯ awsve dev.admin -- dist/chamber-v2.3.3-pre2-5-g059f1d1-dev-linux-amd64 search /servicena  
Service
/servicename/secreta
/servicename/secretb
```

#### TODO

- [ ] Function/Method documentation
- [ ] Any required arguments?
- [ ] Implement S3
- [ ] Tests

pecigonzalo commented:
@nickatsegment sorry to tag you directly but, could you confirm if this is desired? so I can finish the TODO list.

nickatsegment commented:
@pecigonzalo 	No problem at all. Feel free to do so.

A few design questions:
- Are we searching for services or secrets? Your example shows a service name as input, but lists all secrets for that service as output.
- Define "search" :) Looks like from your code it's just looking for a prefix. What about partial matches? We need to make sure it's simple to implement for all backends. (The S3 backend could be pretty dynamic since it's got an index file, but we need to cater to the lowest common subset.)

Also, this sounds very close to https://github.com/segmentio/chamber/issues/141. Maybe instead of calling it "search", it could look like `chamber list-services --prefix=servicena` or `chamber list-services 'servicena*'`

pecigonzalo commented:
@nickatsegment

It would search for a "path", the same thing can be implemented in S3 I believe. Im building on SSM "limitations" as that is the original store, and the API is a bit more limited, I believe it does not support searching for partials AFAIK, but ill try and if possible implement it. We would have to list the entire SSM store and filter client side, but that would be quite ugly and inefficient. 

I was thinking more in the lines of `beginsWith` and just list all "childs", even the secret name, but I can strip that if desired, I thought it might be useful.
Maybe we can just support a couple of search options. 

Regarding #141, yeah, i think it will "fix" that issue.

Given there is an appetite for it, ill start doing a proper implementation supporting this.

pecigonzalo commented:
Just FYI, this is still in my radar, I just did not have the time to finish it.

pecigonzalo commented:
@nickatsegment renamed and implemented a bit better for SSM, tests still missing.
I was trying to implement the S3 one, but given the current store, implementation is experimental and makes a bit hard to look for secrets. 
I can do a `ListObjects` recursively and filter the `__latest.json` files from the output. What do you think?


For the time being, I implemented SSM and put a notice for S3, in case you want to merge this with SSM only.

nickatsegment commented:
Thanks for following through @pecigonzalo !

osterman commented:
Thanks @pecigonzalo !! Excited for this enhancement.
